.pre-commit-config.yaml

```diff
@@ -0,0 +1,8 @@
+repos:
+  - repo: local
+    hooks:
+      - id: poetry-checks
+        name: Run pre-commit checks (format, sort-imports, check-mypy)
+        entry: bash -c 'poetry run format && poetry run sort-imports && poetry run check-mypy'
```

Comment:
Did the action version of what are current lint does 
![image](https://github.com/user-attachments/assets/119c4fc7-565e-4076-b446-84bc9a20d011)